sdk/couchbase/src/options/query_options.rs

```diff
@@ -118,9 +119,25 @@ impl QueryOptions {
         self
     }
 
-    pub fn named_parameters(mut self, named_parameters: HashMap<String, Value>) -> Self {
-        self.named_parameters = Some(named_parameters);
-        self
+    pub fn add_named_parameter<T: Serialize>(
+        mut self,
+        key: impl Into<String>,
+        value: T,
+    ) -> error::Result<Self> {
+        let value = serde_json::to_value(&value)?;
+
+        match self.named_parameters {
```

Comment:
nit: could do: 

```
if let Some(mut params) = self.named_params {
...
} else {
...
}
```

Reply:
Interestingly clippy isn't highlighting that as a recommendation